### ISSUE TITLE
mkl_lapack32 and mkl_lapack64 are obsolete

### ISSUE THREAD

**Issue by davidedelvento** [numpy/numpy#6704]
With newer versions of the intel compiler and its corresponding mkl library, the -mkl_lapack32 and -mkl_lapack64 flags are now obsolete. 

See for example https://mail.scipy.org/pipermail/scipy-user/2012-July/032503.html

The following patch will fix it, however it may break compatibility with older versions of intel compiler and mkl. Since people who use intel compiler and mkl library are performance sensitive, and since intel compiler v16.0.0 improves performance noticeably compared to old versions, I claim that not being a big issue. The following patch will fix it:

```
diff -wu numpy/distutils/system_info.py.orig numpy/distutils/system_info.py
--- numpy/distutils/system_info.py.orig 2015-11-19 16:24:31.253412000 -0700
+++ numpy/distutils/system_info.py      2015-11-19 16:27:46.243694265 -0700
@@ -1039,7 +1039,7 @@
             lapack_libs = self.get_libs('lapack_libs', ['mkl_lapack'])
         else:
             lapack_libs = self.get_libs('lapack_libs',
-                                        ['mkl_lapack32', 'mkl_lapack64'])
+                                        [])

         info = {'libraries': lapack_libs}
         dict_append(info, **mkl)
```

Tested for intel v16.0.0 and MKL v11.3.0

**Comment by charris** [comment]
Could you submit a regular pull request for this?

**Comment by davidedelvento** [comment]
Sure.  
Tomorrow I will test it with Intel v16.0.1 and the corresponding mkl (which have been released since I submitted the report) and submit the pull request.
Cheers,
Davide

**Comment by mattip** [comment]
Fixed in 2088e1c
